core/src/parquet/util/hash_util.rs

```diff
@@ -106,16 +108,12 @@ unsafe fn crc32_hash(bytes: &[u8], seed: u32) -> u32 {
     let num_words = num_bytes / u32_num_bytes;
     num_bytes %= u32_num_bytes;
 
-    let bytes_u32: &[u32] = std::slice::from_raw_parts(
-        &bytes[0..num_words * u32_num_bytes] as *const [u8] as *const u32,
-        num_words,
-    );
```

Comment:
This is unsafe because it does not create a u32-aligned buffer and results in a panic with Rust 1.78